GameServerLib/Logic/Chatbox/Commands/SpawnCommand.cs

```diff
@@ -62,6 +62,42 @@ public override void Execute(Peer peer, bool hasReceivedArguments, string argume
                 _game.Map.AddObject(melee);
                 _game.Map.AddObject(super);
             }
+            else if (split[1] == "minionspurple")
+            {
+                var champion = _playerManager.GetPeerInfo(peer).Champion;
+                var random = new Random();
+
+                var caster = new Minion(MinionSpawnType.MINION_TYPE_CASTER, MinionSpawnPosition.SPAWN_RED_BOT);
+                var cannon = new Minion(MinionSpawnType.MINION_TYPE_CANNON, MinionSpawnPosition.SPAWN_RED_BOT);
+                var melee = new Minion(MinionSpawnType.MINION_TYPE_MELEE, MinionSpawnPosition.SPAWN_RED_BOT);
+                var super = new Minion(MinionSpawnType.MINION_TYPE_SUPER, MinionSpawnPosition.SPAWN_RED_BOT);
```

Comment:
This whole function should be re-done to avoid duplicate code. One way to do this would be as follows:
```CS
public override void Execute(Peer peer, bool hasReceivedArguments, string arguments = "")
{
    var _game = Program.ResolveDependency<Game>();
    var _playerManager = Program.ResolveDependency<PlayerManager>();

    var args = arguments.ToLower().Split(' ');

    if (args.Length < 2)
    {
        _owner.SendDebugMsgFormatted(DebugMsgType.SYNTAXERROR);
        ShowSyntax();
        return;
    }

    var minionTypes = new MinionSpawnType[] {
        MinionSpawnType.MINION_TYPE_CASTER,
        MinionSpawnType.MINION_TYPE_CANNON,
        MinionSpawnType.MINION_TYPE_MELEE,
        MinionSpawnType.MINION_TYPE_SUPER
    };
    MinionSpawnPosition[] minionSpawns;

    var team = "both"
    if (args.Length >= 3)
    {
        team = args[2]
    }

    if (team == "red")
    {
        minionSpawns = new MinionSpawnPosition[] {
            MinionSpawnPosition.SPAWN_RED_BOT
        };
    }
    else if (team == "blue")
    {
        minionSpawns = new MinionSpawnPosition[] {
            MinionSpawnPosition.SPAWN_BLUE_BOT
        };
    }
    else if (team == "both")
    {
        minionSpawns = new MinionSpawnPosition[] {
            MinionSpawnPosition.SPAWN_BLUE_BOT,
            MinionSpawnPosition.SPAWN_RED_BOT
        };
    }

    if (args[1] == "minions")
    {
        var champion = _playerManager.GetPeerInfo(peer).Champion;
        var random = new Random();
        const int randomRange = 400;

        foreach(var spawnLocation in minionSpawns)
        {
            foreach(var minionType in minionTypes)
            {
                var minion = new Minion(minionType, spawnLocation);
                minion.setPosition(
                    champion.X + random.next(-randomRange, randomRange),
                    champion.Y + random.next(-randomRange, randomRange)
                );
                minion.PauseAI(true);

                var waypoint = new Vector2(minion.X, minion.Y);
                minion.SetWaypoints(new List<Vector2> { waypoint, waypoint });

                var team = Enet.TeamId.TEAM_BLUE;
                if(spawnLocation == MinionSpawnPosition.SPAWN_RED_BOT)
                {
                    team = Enet.TeamId.TEAM_PURPLE;
                }
                minion.SetVisibleByTeam(team, true);
                _game.Map.AddObject(minion);
            }
        }
    }
}
```